platform/default/mbgl/storage/offline_download.cpp

```diff
@@ -253,23 +253,40 @@ void OfflineDownload::ensureResource(const Resource& resource,
     *workRequestsIt = util::RunLoop::Get()->invokeCancellable([=]() {
         requests.erase(workRequestsIt);
 
-        optional<std::pair<Response, uint64_t>> offlineResponse =
-            offlineDatabase.getRegionResource(id, resource);
-        if (offlineResponse) {
-            if (callback) {
-                callback(offlineResponse->first);
-            }
+        if (callback) {
```

Comment:
Can you refactor this change so that the common code in each branch of this conditional is not repeated? E.g.:

```c++
auto getResourceSizeInDatabase = [&] () -> optional<uint64_t> {
    if (!callback) {
        return offlineDatabase.hasRegionResource(id, resource);
    }
    optional<std::pair<Response, uint64_t>> response =
        offlineDatabase.getRegionResource(id, resource);
    if (!response) {
        return {};
    }
    callback(response->first);
    return response->second;
}

optional<uint64_t> offlineResponse = getResourceSizeInDatabase();
if (offlineResponse) {
    ...
}
```